cpp/src/parquet/column_writer.cc

```diff
@@ -292,12 +283,20 @@ class SerializedPageWriter : public PageWriter {
     }
 
     format::PageHeader page_header;
-    page_header.__set_type(format::PageType::DATA_PAGE);
     page_header.__set_uncompressed_page_size(static_cast<int32_t>(uncompressed_size));
     page_header.__set_compressed_page_size(static_cast<int32_t>(output_data_len));
-    page_header.__set_data_page_header(data_page_header);
     // TODO(PARQUET-594) crc checksum
 
+    if (page.type() == PageType::DATA_PAGE) {
+      const DataPageV1& v1_page = dynamic_cast<const DataPageV1&>(page);
```

Comment:
Use `checked_cast` here

Reply:
Good tip, I wasn't aware of that utility.